[Muhammad Arshad Shaik] iP

FireRadical22 opened the pull request:
# **Duke**
### Duke helps you remember your tasks for the day so you don't have to!!

Duke is:
- Text-based
- Simple to interact with
- ~~FAST~~ F1-FAST to use 🏎️ 🏎️ 

All you need to is:
1. Download from [here](https://github.com/FireRadical22/ip/releases/tag/A-Jar)
2. Open Duke by double-clicking on it
3. Start adding your task!
4. Sit back and let Duke do the managing for you! 😎 

And the best part? It's completely **FREE**! 🤩 

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

FireRadical22 committed:
Level-1

FireRadical22 committed:
Level 2

FireRadical22 committed:
Level-3

- Updated Duke Class to mark/unmark task
- Added Task class

FireRadical22 committed:
Level-4

- Updated Duke to accommodate different types of tasks

- Added todo task
- Added deadline task
- Added events task

FireRadical22 committed:
A-TextUiTesting

- Updated runtest.bat with appropriate classpaths
- Updated input.txt and EXPECTED.txt with sample test and sample result respectively
- Cleaning code for mark and unmark command

FireRadical22 committed:
Level-5

- Update Duke to handle user input errors
- Create DukeException class to be thrown in Duke class

FireRadical22 committed:
Level-6

- Added delete feature into Duke

FireRadical22 committed:
Code cleanup

- Created mark and unmark method to throw DukeException in the event of user input error
- Updated delete method to throw DukeException in the event of user input error

FireRadical22 committed:
Level-7

- Added Save feature (temp before merge)

FireRadical22 committed:
Level-8 Test

- Mandate use of proper date format for deadline and event commands (in testing)

FireRadical22 committed:
Level-7 Update

- Update Duke program to check file path in OS-independent way
- Included write or read file feature for appropriate commands

FireRadical22 committed:
Merge branch 'branch-Level-7'

FireRadical22 committed:
Level-7 Fix

- Fix new lines not being generated when a new Task is added

FireRadical22 committed:
Merge branch 'branch-Level-8'

FireRadical22 committed:
Level-8 Fix

- Add LocalDateTime to store date and time for Deadline and Event objects
- Update checking of deadline, todo and event commands

FireRadical22 committed:
Apply more OOP to program

- Add Command class to execute commands
- Add UI class for UI feature
- Add Parser class to Pass input from prompt
- Add Storage class to store and access file from file path.
- Add TaskList class to keep track of tasks.
- Update Duke class to implement use of new classes

FireRadical22 committed:
Package classes to respective packages

- Create package duke.task for Task classes and duke for non-Task classes

FireRadical22 committed:
Merge branch 'add-gradle-support'

FireRadical22 committed:
Update build.gradle

- Update mainClassName

FireRadical22 committed:
Add JUnit Testing

- Add JUnit5 Test cases
- Fix bugs in Duke program

FireRadical22 committed:
Update Duke program and DukeTest

- Add more exception handling when using TaskList and Command classes
- Update DukeTest class to check for correctness of command inputs.

FireRadical22 committed:
Add JavaDoc

- Add JavaDoc Documentation for classes and methods

FireRadical22 committed:
Merge branch 'branch-A-JavaDoc'

FireRadical22 committed:
Adhere to coding standard

- Edit classes and their methods to comply with coding standard

FireRadical22 committed:
Merge branch 'branch-A-CodingStandard'

FireRadical22 committed:
Add find feature

FireRadical22 committed:
Merge branch 'branch-Level-9'

bernicetoh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,206 @@
+package duke;
+
+import duke.task.*;
```
Should the imported classes be listed explicitly?

FireRadical22 replied:
Hi, i used * as in this case, i needed all of the classes within the duke.task package, which were Task, ToDo, Deadline and Events classes to handle storage of different Task objects.

bernicetoh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,206 @@
+     * @param argument key word or key phrase to find
+     * @return list of strings that contains key word or key phrase
+     */
+    public List<String> getDescMatches(String argument) {
```
```suggestion
    public List<String> getMatchingDesc(String argument) {
```

bernicetoh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+    private LocalDateTime parseDateTime(String deadline) {
+        String[] dateAndTime = deadline.split(" ");
+        String[] dayMonthYear = dateAndTime[0].split("/");
+        int hour = Integer.parseInt(dateAndTime[1])/100;
```
```suggestion
        int hour = Integer.parseInt(dateAndTime[1]) / 100;
```

bernicetoh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+        String[] dateAndTime = deadline.split(" ");
+        String[] dayMonthYear = dateAndTime[0].split("/");
+        int hour = Integer.parseInt(dateAndTime[1])/100;
+        int minute = Integer.parseInt(dateAndTime[1])%100;
```
```suggestion
        int minute = Integer.parseInt(dateAndTime[1]) % 100;
```

bernicetoh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+        int hour = Integer.parseInt(dateAndTime[1])/100;
+        int minute = Integer.parseInt(dateAndTime[1])%100;
+        int[] dateNumbers = new int[3];
+        for (int i = 0; i < dayMonthYear.length;i++) {
```
```suggestion
        for (int i = 0; i < dayMonthYear.length; i++) {
```

bernicetoh commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,61 @@
+    private LocalDateTime parseDateTime(String period) {
+        String[] dateAndTime = period.split(" ");
+        String[] dayMonthYear = dateAndTime[0].split("/");
+        int hour = Integer.parseInt(dateAndTime[1])/100;
```
```suggestion
        int hour = Integer.parseInt(dateAndTime[1]) / 100;
```

bernicetoh commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,61 @@
+        String[] dateAndTime = period.split(" ");
+        String[] dayMonthYear = dateAndTime[0].split("/");
+        int hour = Integer.parseInt(dateAndTime[1])/100;
+        int minute = Integer.parseInt(dateAndTime[1])%100;
```
```suggestion
        int minute = Integer.parseInt(dateAndTime[1]) % 100;
```

FireRadical22 replied:
Hi, thanks for pointing out the mistakes. I've made the changes based on your suggestion. 😄

bernicetoh reviewed (commented):
Overall, your code is neat and easy to read. There are a few instances where you could have added space characters between operators.

FireRadical22 committed:
Update src/main/java/duke/TaskList.java

- Update method name for getting descriptions of matching task

Co-authored-by: bernicetoh <87461298+bernicetoh@users.noreply.github.com>

FireRadical22 committed:
Update code to adhere to coding standard

- Changes to code suggested by bernicetoh

pzaiming commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,226 @@
+            throw new DukeException("What is the Event task???");
+        } else {
+            Pattern eventPattern = Pattern
+                    .compile(".+/from \\d{2}/\\d{2}/\\d{4} \\d{4} /to \\d{2}/\\d{2}/\\d{4} \\d{4}");
```
Great use of regex right here!

pzaiming commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,226 @@
+            for (String desc : descOfTasksFound) {
+                System.out.println(desc);
+            }
+            System.out.println("_____________________________________\n");
```
You may want to create a method in the UI class to print out these lines here.

pzaiming commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,206 @@
+    }
+
+    /**
+     * stores Event tasks into taskList
```
Capitalization should be used for consistency but otherwise, your docs written are clear and concise!

pzaiming commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,226 @@
+     * @throws DukeException when command cannot be executed due to incorrect format
+     */
+    public void execArgs(TaskList taskList) throws DukeException {
+        switch(this.command.toLowerCase()) {
```
Great usage of .toLowercase() to ensure more flexibility in the command.

pzaiming reviewed (commented):
Great job! Code is well written and done naturally with OOP.

FireRadical22 committed:
Creation of Duke Application

- Integrate JavaFX features into program

FireRadical22 committed:
Add Assertions into program

- Add assertions where appropriate in program

FireRadical22 committed:
Fix Bugs in Duke Program

- Update "bye" command to exit program automatically
- Clear userInput box after everywhere command inputted
- Update error and action messages

FireRadical22 committed:
-Remove old methods in Duke

FireRadical22 committed:
Improve Code Quality of program

- Add further abstractions to some methods
- Remove any use of magic Strings or numbers
- Fix any methods where SLAP is weak
- Add comments where necessary
- Simplify expressions, eliminate deep nesting and long methods where necessary

FireRadical22 committed:
Revert "Improve Code Quality of program"

This reverts commit 80c97a9a1b7cac9abe9d27932e104cbc397eb1ef.

FireRadical22 committed:
Revert "Revert "Improve Code Quality of program""

This reverts commit 51eab33c4aa015c71bc3c755c2a85d6200ae9858.

FireRadical22 committed:
Improved Code Quality a bit more

FireRadical22 committed:
Add Tagging feature

Implement Tagging feature to allow users to create custom categories for their tasks based on their needs

Duke has been relying on a terminal to get input from the user. As such, better user interface is needed to improve user's experience with Duke program.

Replace UI class with GUI class, to integrate JavaFX into Duke as Duke will evolve to an application with command-line interface

FireRadical22 committed:
Merge pull request #2 from FireRadical22/branch-C-Tagging

Add Tagging feature

FireRadical22 committed:
Change code organisation

Move appropriate classes into their respective packages for better organisation

Add MainWindow class to handle main window of Duke Application. This is to address the problem of infinite recursion caused by initialisation of Duke object in GUI class and GUI object in Duke class.

FireRadical22 committed:
Update DukeTest.java

Delete sampleTest used for testing out certain classes in Java 11 API. sampleTest is no longer needed as it does not test the program itself.

FireRadical22 committed:
Update README.md

Add Duke's features, command formats and short description of the features
Add Illustrations and examples of use of features in later commits

FireRadical22 committed:
Add images for UG and Fix minor bugs

Add images for User Guide to illustrate how different features of Duke works

Fix minor bugs found to ensure Duke program runs as intended

FireRadical22 committed:
Update README.md

Add Image for respective features

FireRadical22 committed:
Adjust photo

Adjust screenshot for deadline feature

FireRadical22 committed:
Merge branch 'master' of https://github.com/FireRadical22/ip

FireRadical22 committed:
Update README.md

Update images for features as images were initially not being shown
Give links for features for better navigability

FireRadical22 committed:
Update gradle build and classes

FireRadical22 committed:
Update build.gradle

FireRadical22 committed:
Update README.md

Add Back To Top link for better navigability in README
Add summary to summary all features of Duke Program

FireRadical22 committed:
Create UI.PNG

FireRadical22 committed:
Rename file

Rename file to UI.png for tracking by iP progress dashboard

FireRadical22 committed:
Move UI.png

FireRadical22 committed:
Move and Rename UI image file

FireRadical22 committed:
Rename Ui.PNG
